src/hotspot/share/interpreter/interpreterRuntime.cpp

```diff
@@ -1156,6 +1156,16 @@ void InterpreterRuntime::cds_resolve_invokedynamic(int raw_index,
   pool->cache()->set_dynamic_call(info, raw_index);
 }
 
+JRT_ENTRY(void, InterpreterRuntime::end_training_check(JavaThread* current)) {
+  LastFrameAccessor last_frame(current);
+  ResourceMark rm(current);
+  methodHandle m (current, last_frame.method());
+  if(m->is_end_training_trigger()) {
```

Comment:
unlike c1 and c2, we need to check the method flag here, as the generated code is used for many methods and so has to be included in all interpreter methods